### ISSUE TITLE
📎 Allow empty type parameter list for interfaces and type aliases

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1237]
### Description

TypeScript allows interface declarations and type aliases to have empty type parameter lists. Biome parser doesn't allow empty type parameter lists.

```ts
interface Foo<> {}
type Bar<> = {}
```

[TypeScript Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt0B4B8yDeAvgFBgCeADigEJxS7IC8BhQA)
[Biome playground](https://biomejs.dev/playground/?enabledLinting=false&code=aQBuAHQAZQByAGYAYQBjAGUAIABGAG8AbwA8AD4AIAB7AH0ACgB0AHkAcABlACAAQgBhAHIAPAA%2BACAAPQAgAHsAfQA%3D)

Biome should accept empty type parameter lists from interfaces and type aliases.

This was [recently fixed in ESbuil](https://github.com/evanw/esbuild/pull/3513).

**Comment by ematipico** [comment]
I wonder why such a thing is allowed

**Comment by Conaclos** [comment]
> I wonder why such a thing is allowed

Yeah, I dream of a simpler version of TypeScript... Flow forbids empty type parameter lists.

Once fixed, we should add a linter rule `noEmptyTypeParameters`.

**Comment by togami2864** [comment]
I'll take a look this later
